app/ui-login/client/reset-password/resetPassword.js

```diff
@@ -27,25 +27,23 @@ Template.resetPassword.helpers({
 	},
 });
 
-const resetPassword = (token, password) => new Promise((resolve, reject) => {
-	Accounts.resetPassword(token, password, function(error, result) {
-		if (!error) {
-			FlowRouter.go('home');
-			toastr.success(t('Password_changed_successfully'));
-			callbacks.run('userPasswordReset');
-			resolve(result);
-		}
-
+async function resetPassword(token, password) {
+	try {
+		await call('checkUserPassword', password);
+		Accounts.resetPassword(token, password);
```

Comment:
`Accounts.resetPassword` does not fire errors synchronously, it's necessary to keep the callback